backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/impl/PaymentRequestDAO.java

```diff
@@ -22,39 +22,55 @@
  * #L%
  */
 
-
-import java.util.HashSet;
-import java.util.List;
-import java.util.Properties;
-import java.util.Set;
-
+import de.metas.banking.BankAccountId;
+import de.metas.banking.model.I_C_Payment_Request;
+import de.metas.banking.payment.IPaymentRequestDAO;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import lombok.NonNull;
 import org.adempiere.ad.dao.IQueryBL;
 import org.adempiere.ad.dao.IQueryBuilder;
 import org.adempiere.ad.trx.api.ITrx;
+import org.adempiere.model.InterfaceWrapperHelper;
 import org.compiere.model.I_C_Invoice;
+import org.jetbrains.annotations.Nullable;
 
-import de.metas.banking.model.I_C_Payment_Request;
-import de.metas.banking.payment.IPaymentRequestDAO;
-import de.metas.util.Check;
-import de.metas.util.Services;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
 
 /**
  * @author al
  */
 public class PaymentRequestDAO implements IPaymentRequestDAO
 {
-	private final IQueryBuilder<I_C_Payment_Request> retrieveRequestForInvoiceQuery(final I_C_Invoice invoice)
+	private IQueryBuilder<I_C_Payment_Request> retrieveRequestForInvoiceQuery(final I_C_Invoice invoice)
 	{
 		final IQueryBL queryBL = Services.get(IQueryBL.class);
 
-		final IQueryBuilder<I_C_Payment_Request> queryBuilder = queryBL.createQueryBuilder(I_C_Payment_Request.class, invoice)
+		return queryBL.createQueryBuilder(I_C_Payment_Request.class, invoice)
 				.addEqualsFilter(I_C_Payment_Request.COLUMN_C_Invoice_ID, invoice.getC_Invoice_ID())
 				.addOnlyActiveRecordsFilter() // NOTE: very important, BL relies on this (i.e. assumes it gets only active requests)
 				.addOnlyContextClient();
-		return queryBuilder;
 	}
 
 	@Override
+	public void createOrReplace(@NonNull final I_C_Invoice invoice, @NonNull final BankAccountId bankAccountId)
```

Comment:
consider `InvoiceId invoiceId` as param, instead of invoice.

retrieveRequestForInvoiceQuery can simply work with InvoiceId and use the current ctx/thread inherited trx.